examples/google-cloud/cloud-functions/go/visit.go

```diff
@@ -0,0 +1,325 @@
+// vespa visit command
+// Author: arnej, kkraune
+
+package vespasamples
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/functions-framework-go/functions"
+)
+
+func init() {
+	functions.HTTP("visit", visit)
+}
+
+func ReaderToJSON(reader io.Reader) string {
+	bodyBytes, _ := io.ReadAll(reader)
+	var prettyJSON bytes.Buffer
+	parseError := json.Indent(&prettyJSON, bodyBytes, "", "    ")
+	if parseError != nil { // Not JSON: Print plainly
+		return string(bodyBytes)
+	}
+	return prettyJSON.String()
+}
+
+// Service represents a Vespa service.
+type Service struct {
+	BaseURL string
+	Name    string
+	Cert    tls.Certificate
+}
+
+type OperationResult struct {
+	Success bool
+	Message string // Mandatory message
+	Detail  string // Optional detail message
+	Payload string // Optional payload - may be present whether or not the operation was success
+}
+
+func Success(message string) OperationResult {
+	return OperationResult{Success: true, Message: message}
+}
+
+func Failure(message string) OperationResult {
+	return OperationResult{Success: false, Message: message}
+}
+
+func FailureWithPayload(message string, payload string) OperationResult {
+	return OperationResult{Success: false, Message: message, Payload: payload}
+}
+
+//
+//
+//
+
+func visit(w http.ResponseWriter, r *http.Request) {
+	var d struct {
+		ContentCluster string `json:"contentCluster"`
+		Selection      string `json:"selection"`
+		ChunkCount     int    `json:"chunkCount"`
+		Endpoint       string `json:"endpoint"`
+	}
+	if err := json.NewDecoder(r.Body).Decode(&d); err != nil {
+		fmt.Fprint(w, "Visit: Error decoding arguments")
+		return
+	}
+	if d.Endpoint == "" {
+		fmt.Fprint(w, "Visit: endpoint missing")
+		return
+	}
+	if d.ContentCluster == "" {
+		fmt.Fprint(w, "Visit: contentCluster missing")
+		return
+	}
+	intro := fmt.Sprintf("Visit - Args: %s, %s, %s, %d\n",
+		d.Endpoint,
+		d.ContentCluster,
+		d.Selection,
+		d.ChunkCount)
+	fmt.Fprintf(w, intro)
+	log.Printf(intro)
+
+	var vArgs visitArgs
+	vArgs.contentCluster = d.ContentCluster
+	if d.Selection != "" {
+		vArgs.selection = d.Selection
+	}
+	if d.ChunkCount != 0 {
+		vArgs.chunkCount = d.ChunkCount
+	}
+
+	certPem := []byte(os.Getenv("SEC_CERT"))
+	keyPem := []byte(os.Getenv("SEC_KEY"))
+	cert, err := tls.X509KeyPair(certPem, keyPem)
+	if err != nil {
+		fmt.Fprintf(w, "Cert error: %v", err)
+		return
+	}
+
+	var service Service
+	service.BaseURL = d.Endpoint //"https://vespacloud-docsearch.vespa-team.aws-eu-west-1a.z.vespa-app.cloud"
+	service.Cert = cert
+
+	res := visitClusters(vArgs, &service)
+
+	fmt.Fprintf(w, "Msg: %s\n", res.Message)
+}
+
+type visitArgs struct {
+	contentCluster string
+	fieldSet       string
+	selection      string
+	makeFeed       bool
+	jsonLines      bool
+	pretty         bool
```

Comment:
I think it's best to remove streamFlag, chunkTime and maybe pretty from this code.

Reply:
Agreed! will clean up